destination/config.go

```diff
@@ -113,13 +113,20 @@ func (c Config) Validate() error {
 		multierr = append(multierr, err)
 	}
 
+	if c.RecordFormat != nil {
```

Comment:
In the previous PR, this part was put in the Open() method: https://github.com/ConduitIO/conduit-connector-kafka/pull/180/files#diff-b0e314d1e24c22a62ecb712e94913d77bed15abfd75e80a0b61189c0abe17453R42.

However, I think after that we were talking about custom parsing and that it should happen in Validate(), so I've put it here.